File: CodeForces/1601-1700/1624/A/1624A.py
# Problem 1624A: Plus One on the Subset
# https://codeforces.com/contest/1624/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 27th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

runs = int(input())

for _ in range(0,runs):
    arr_size = int(input())
    vals = sorted(list(map(int, input().split())))
    print(vals[arr_size-1] - vals[0])

File: CodeForces/1601-1700/1624/A/1624A.cs
// Problem 1624A: Plus One on the Subset
// https://codeforces.com/contest/1624/problem/A
// Compiled with: Mono JIT compiler version 6.12.0.199 (tarball Fri May  2 12:22:51 UTC 2025)
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 26th, 2025
// Codeforces language used: C# Mono 6.8

using System;
using System.Linq;

class Program {
    static void Main() {
        int runs = int.Parse(Console.ReadLine());
        for (int i = 0; i < runs; i++) {
            int count = int.Parse(Console.ReadLine());
            int[] all_nums = Console.ReadLine().Split().Select(int.Parse).OrderBy(x => x).ToArray();
            Console.WriteLine(all_nums[count-1]-all_nums[0]);
        }
    }
}
